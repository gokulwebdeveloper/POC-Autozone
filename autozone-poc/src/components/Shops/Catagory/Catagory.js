import React, { Fragment } from "react";
import ListItem from "../ListItem/ListItem";
import PropTypes from 'prop-types';


Catagory.propTypes = {
    data: PropTypes.string,
    data: PropTypes.any.isRequired,
}

export default function Catagory(props) {
    if (props != undefined && props.data != undefined) {
        return (
            <Fragment>
                <li data-toggle="collapse" data-target={props.datatarget}>
                    <a href="#">{props.data.title}</a>
                    <ul className="sub-menu collapse hide" id={props.id}>
                        <ListItem items={props.data.items} />
                    </ul>
                </li>
            </Fragment>
        );
    } else {
        return <p>-</p>
    }

}