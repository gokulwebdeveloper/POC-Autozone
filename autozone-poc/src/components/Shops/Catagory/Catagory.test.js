import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import React, { Component } from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { render } from '@testing-library/react';
import Catagory from './Catagory';

configure({ adapter: new Adapter() });


describe('Catagory testing', () => {
    var categoryList = [{
		"title": "clothing",
		"items": [
			"All",
			"Bodysuits",
			"Dresses",
			"Hoodies Sweats",
			"Jackets Coats",
			"Jeans",
			"Pants Leggings",
			"Rompers Jumpsuits",
			"Shirts Blouses",
			"Shirts",
			"Sweaters &amp; Knits"
		]
	}
];


    let wrapper;
    wrapper = mount( <Catagory data={categoryList[0]} datatarget="#clothing" id="clothing"/>);

    it("SnapShot Match for Catagory component", () => {
        const page = renderer.create( <Catagory data={categoryList[0]} datatarget="#clothing" id="clothing"/>).toJSON();
        expect(page).toMatchSnapshot();
      });

	  test('renders ', () => {
        expect(wrapper.find('li').at(2).text()).toBe('Bodysuits');
    });

  })