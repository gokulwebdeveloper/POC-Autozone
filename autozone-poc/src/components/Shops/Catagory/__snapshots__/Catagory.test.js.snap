// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catagory testing SnapShot Match for Catagory component 1`] = `
<li
  data-target="#clothing"
  data-toggle="collapse"
>
  <a
    href="#"
  >
    clothing
  </a>
  <ul
    className="sub-menu collapse hide"
    id="clothing"
  >
    <li>
      <a
        href="#"
      >
        All
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Bodysuits
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Dresses
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Hoodies Sweats
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Jackets Coats
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Jeans
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Pants Leggings
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Rompers Jumpsuits
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Shirts Blouses
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Shirts
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Sweaters &amp; Knits
      </a>
    </li>
  </ul>
</li>
`;
